apps/src/javalab/Javalab.js

```diff
@@ -235,6 +247,12 @@ Javalab.prototype.init = function(config) {
     }
   }
 
+  // If we're editing an exemplar, override initial sources
+  // with the exemplar code saved to the level definition.
+  if (this.isEditingExemplar && config.level.exemplarSources) {
+    getStore().dispatch(setAllSources(config.level.exemplarSources));
```

Comment:
If there are no exemplarSources, does it make sense to set this to the startSources?

Reply:
Oh wait, that's what's happening in the section above here. Nevermind.

Reply:
Yeah I got a bit confused by this as well -- I feel It might make sense to move a lot of this logic to the Rails side and just send some property like `initialSources` that Javalab can use.

Reply:
This feels a little fragile. If someone is tidying this code up and moves this block of code above the previous block (i.e. the start sources block), exemplar mode will break because the user's code will always overwrite the exemplar code. This could be solved by moving this to the first block in an `if`/`else if`. i.e.
```
if (this.isEditingExemplar && config.level.exemplarSources) {
  // use exemplar sources
else if (
    startSources &&
    typeof startSources === 'object' &&
    Object.keys(startSources).length > 0
) {
  // use the code stored in the startSources variable
}

Reply:
Yeah I think Rails would be a longer term thing, what you suggested sounds good for now.